halo2_backend/src/plonk/prover.rs

```diff
@@ -410,40 +459,63 @@ impl<
             }
         }
 
+        // Move on
+
         self.next_phase_index += 1;
         Ok(challenges.clone())
     }
 
     /// Finalizes the proof creation.
+    /// The following steps are performed:
+    /// - 1. Generate commited lookup polys
+    /// - 2. Generate commited permutation polys
+    /// - 3. Generate commited lookup polys
+    /// - 4. Generate commited shuffle polys
+    /// - 5. Commit to the vanishing argument's random polynomial
+    /// - 6. Generate the advice polys
+    /// - 7. Evaluate the h(X) polynomial
```

Comment:
This is confusing for me (even if this is a reference to the original code, which has the function `evaluate_h`, and there was even a comment saying `Evaluate the h(X) polynomial`).

The way I see it, we're not evaluating `h(X)`, instead we're calculating `h(X)`.  I say this because in my mind, if we evaluate a polynomial, the result is a field value.  But the result of step `7` is a polynomial.